netmanager-app/core/hooks/useCities.ts

```diff
@@ -0,0 +1,28 @@
+import { useQuery } from "@tanstack/react-query";
+import { useDispatch } from "react-redux";
+import { cities } from "../apis/cities";
+import { setCities, setError } from "../redux/slices/citiesSlice";
+import { useAppSelector } from "../redux/hooks";
+
+export const useCities = () => {
+  const dispatch = useDispatch();
```

Comment:
Lets call the hook useGrids and the corresponding apis and redux states call them also grids. This not only returns data for cities but for countries too and other kinds of administrative geographical levels @danielmarv